core/src/avm1/value.rs

```diff
@@ -271,20 +273,35 @@ impl<'gc> Value<'gc> {
         &self,
         other: Value<'gc>,
         activation: &mut Activation<'_, 'gc, '_>,
-    ) -> Result<Option<bool>, Error<'gc>> {
+    ) -> Result<Value<'gc>, Error<'gc>> {
+        // If either parameter's `valueOf` results in a non-movieclip object, immediately return false.
+        // This is the common case for objects because `Object.prototype.valueOf` returns the same object.
+        // For example, `{} < {}` is false.
         let prim_self = self.to_primitive_num(activation)?;
+        if matches!(prim_self, Value::Object(o) if o.as_display_object().is_none()) {
+            return Ok(false.into());
```

Comment:
Seems like if both operands are plain objects, then both `valueOf` callbacks should be invoked:
```as
var a = { valueOf: function() { trace("in a.valueOf"); } };
var b = { valueOf: function() { trace("in b.valueOf"); } };
trace(a < b); // traces "a.valueOf", "b.valueOf"
trace(b < a); // traces "b.valueOf", "a.valueOf"
```
If I get it right, Ruffle with this PR would bail-out early and invoke only the first `valueOf`.
So I think the correct order is something like this:
1. `a.to_primitive_num()`
2. `b.to_primitive_num()`
3. if `prim_self` is an object, return `false`
4. if `prim_other` is an object, return `false`

Reply:
The PR runs this example correctly -- the valueOfs in this case return `undefined` and not an Object, so the early-exit doesn't happen.

The early-exit does happen if you add `return {}` to the valueOf calls.